macosx/Array+Additions.swift

```diff
@@ -0,0 +1,26 @@
+// This file Copyright © 2022-2023 Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+import Foundation
+
+extension Array {
```

Comment:
This is such a good candidate for refactoring in the future, maybe let's leave a TODO

Reply:
Then there would be TODO for everything. Let's leave the freedom of refactoring to those with time to review the code instead of cluttering with TODO that could be unclear and would potentially stay there for years. For now, I'm staying close to the current ObjC implementation.